main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -408,4 +408,52 @@ object RedundancyError {
          |""".stripMargin
     })
   }
+
+  /**
+    * An error raised to indicate that a pure expression value was discarded.
+    *
+    * @param loc the location of the expression.
+    */
+  case class DiscardedPureValue(loc: SourceLocation) extends RedundancyError {
```

Comment:
What's the example here?

Reply:
This would be caught by the other redundancy check related to Stm
```
discard 2;
println("hey")
```
but this would not
```
let x: Unit = discard 2;
println(x)
```